merlin-server/sql/merlin/tables/activity.sql

```diff
@@ -83,3 +95,41 @@ create trigger increment_revision_on_delete_activity_trigger
 after delete on activity
 for each row
 execute function increment_revision_on_delete_activity();
+
+create or replace function generate_activity_name()
+returns trigger
+security definer
+language plpgsql as $$begin
+  execute 'create sequence if not exists activity_name_' || new.plan_id || '_' || new.type;
+  new.name = new.type || ' ' || nextval('activity_name_' || new.plan_id || '_' || new.type);
+  return new;
+end$$;
+
+comment on function generate_activity_name() is e''
+  'Generates a name for an activity as the activity type + number from sequence of the type scoped to its plan.';
+
+create trigger generate_name_trigger
+before insert on activity
+for each row execute function generate_activity_name();
+
+comment on trigger generate_name_trigger on activity is e''
+  'Generates a name for an activity as the activity type + number from sequence of the type scoped to its plan.';
```

Comment:
More of an understanding question:

This is kinda confusing me. What is a sequence with respect to an activity?  Is it something with the Schedular?

Reply:
Sequence as in a Postgres Sequence ie a numerically incrementing series of integers.

Reply:
In this case, it means that for a given plan, each activity directive with the same activity type will have the name `<activityType> 1`, `<activityType> 2`, `<activityType> 3`, etc

Reply:
Ahhh thank for the clarification.